### PR TITLE
Add Celadon Yggrete Shard pop NMs to Ceizak Battlegrounds

### PR DESCRIPTION
**Pull request by Yuchemey**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

"It allows you to spawn the NMs (Notorious Monsters) Supernal Chapuli, Mastop, Unfettered Twitherym, Transcendent Scorpion, and Tax'et in Ceizak Battlegrounds." 

## Steps to test these changes

To achieve this, you need to trade the following items at specific locations in Ceizak Battlegrounds:

Celadon Yggrete Shard I to the ??? at J-7 for Unfettered Twitherym.
Celadon Yggrete Shard II to the ??? at J-9 for Supernal Chapuli.
Celadon Yggrete Shard III to the ??? at I-10 for Transcendent Scorpion.
Celadon Yggrete Shard IV to the ??? at G-8 for Mastop.
Celadon Yggrete Shard V to the Blank Space at G-10 for Tax'et.

### CODE REVIEW

File: scripts/zones/Ceizak_Battlegrounds/npcs/qm_mastop.lua [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+entity.onTrigger = function(player, npc)
+end
+
+entity.onEventUpdate = function(player, csid, option, npc)
+end
+
+entity.onEventFinish = function(player, csid, option, npc)
+end
+
+return entity
```
Comment (Xaver-DaRed) [inline scripts/zones/Ceizak_Battlegrounds/npcs/qm_mastop.lua]: need a newline at the end of the file. Same for all the files below.

File: scripts/zones/Ceizak_Battlegrounds/npcs/qm_mastop.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+-- Spawns Mastop
+-- !pos -230 0 36 261
+-----------------------------------
+local ID = zones[xi.zone.CEIZAK_BATTLEGROUNDS]
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    if
+        npcUtil.tradeHas(trade, xi.item.CELADON_YGGRETE_SHARD_IV) and
```
Comment (Xaver-DaRed) [inline scripts/zones/Ceizak_Battlegrounds/npcs/qm_mastop.lua]: We may want to use `npcUtil.tradeHasExactly()` so players dont put other items in there and get screwed.

**Comment by zach2good** [comment]
Hello and welcome! This all seems reasonable to me. We can squash the PR and rename the commit (like I did with the PR title) on the way in so it's a little more descriptive, or you can do it with a force push - whatever you prefer.

Does this make these NMs into loot pinatas? Or are they hooked up to any other system that means having them poppable but not yet balanced would be a bad thing?

**Comment by zach2good** [comment]
There's some other stuff that needs to be added to these pops that looks like it'll be very easy to add that will mean these don't have to be revisited:

There's some additional messaging:
https://www.youtube.com/watch?v=eQw8711avqM
image

But no additional status effects like a level sync or anything, or confrontation, etc.

**Comment by zach2good** [comment]
and then also, you get plasm in addition to XP, items, etc.

image

**Comment by Yuchemey** [comment]
I thought of starting with pops. then everything that is important, like HP MP Level and the most important off all... their weapon skill i need to datmine in order to find 6 skills missing from them. then imma jump on Drops, Celadon drops. i will do one zone at a time.

**Comment by zach2good** [comment]
Do you have access to retail? You can't typically look into the DATs for what animations are linked to which mobskills, etc. (which is why there are so many holes in mob_skills.sql)

**Comment by zach2good** [comment]
(as long as they're not dropping loot, or letting people cheese anything, these are harmless to go in pretty much as they are)

**Comment by Yuchemey** [comment]
I will put drops at the end when everything is all good and ready. I have access to retail.

**Comment by zach2good** [comment]
> I will put drops at the end when everything is all good and ready.

This is a sane way to handle this, thanks!

For mobskills, you'll need to get captures of the mobs doing the moves you're interested in. Here's a link to the capture discord, which should have instructions for how to get set up to do captures: https://discord.gg/4WsCZf5g

**Review by zach2good — APPROVED** [review]

**Review by Xaver-DaRed — APPROVED** [review]

**Comment by Yuchemey** [comment]
Thanks.

**Review by EpicTaru — APPROVED** [review]

**Review by claywar — COMMENTED** [review]
I'm really excited to see people taking an interest in post-75 content.  That being said, the pop messages are something that needs to be researched, and also seeing that warning makes me wonder if there's a confrontation effect that's not being displayed.  Any time content level is displayed, my spidey-sense says there's something else going on.

**Comment by Yuchemey** [comment]
I did most of the content solo. And solo most of delve content 3 years after realease. I am having fun yet, having trouble with the mechanics. But i will prevail.
________________________________

**Comment by Yuchemey** [comment]
> I'm really excited to see people taking an interest in post-75 content. That being said, the pop messages are something that needs to be researched, and also seeing that warning makes me wonder if there's a confrontation effect that's not being displayed. Any time content level is displayed, my spidey-sense says there's something else going on.

there are 2 ways you get this message. By trigger you get 2 message. "You have a sense that you need to trade <item_name>" and "The Level of difficulty is level 110~116". And you get "The Level of difficulty is level 110~116" By trading item.

I was thinking of making
Popping - PR
Mobskill HP/MP/Level - PR
Then Weakness - PR
Drops + Display Message - PR
then Ceizak is done

**Pull request merged**